contracts/interchain-token-service/InterchainTokenService.sol

```diff
@@ -227,182 +216,99 @@ contract InterchainTokenService is
-    }
-
     /**
      * @notice Used to deploy remote custom TokenManagers.
      * @param salt the salt to be used.
-     * @param destinationChain the name of the chain to deploy the TokenManager and standardized token to.
+     * @param destinationChain the name of the chain to deploy the TokenManager and interchain token to.
      * @param tokenManagerType the type of TokenManager to be deployed.
      * @param params the params that will be used to initialize the TokenManager.
      * @param gasValue the amount of native tokens to be used to pay for gas for the remote deployment. At least
      * the amount specified needs to be passed to the call
      * @dev `gasValue` exists because this function can be part of a multicall involving multiple functions
      * that could make remote contract calls.
      */
-    function deployRemoteCustomTokenManager(
+    function deployTokenManager(
         bytes32 salt,
         string calldata destinationChain,
         TokenManagerType tokenManagerType,
         bytes calldata params,
         uint256 gasValue
-    ) external payable whenNotPaused returns (bytes32 tokenId) {
+    ) external payable whenNotPaused returns (bytes32 tokenId_) {
         address deployer_ = msg.sender;
-        tokenId = customTokenId(deployer_, salt);
-
-        emit CustomTokenIdClaimed(tokenId, deployer_, salt);
+        tokenId_ = tokenId(deployer_, salt);
 
-        _deployRemoteTokenManager(tokenId, destinationChain, gasValue, tokenManagerType, params);
-    }
-
-    /**
-     * @notice Used to deploy a standardized token alongside a TokenManager.
-     * @param salt the salt to be used.
-     * @param name the name of the token to be deployed.
-     * @param symbol the symbol of the token to be deployed.
-     * @param decimals the decimals of the token to be deployed.
-     * @param mintAmount the amount of token to be mint during deployment to msg.sender.
-     * @param distributor the address that will be able to mint and burn the deployed token.
-     */
-    function deployAndRegisterStandardizedToken(
-        bytes32 salt,
-        string calldata name,
-        string calldata symbol,
-        uint8 decimals,
-        uint256 mintAmount,
-        address distributor
-    ) external payable whenNotPaused {
-        bytes32 tokenId = customTokenId(msg.sender, salt);
-        _deployStandardizedToken(tokenId, distributor, name, symbol, decimals, mintAmount, msg.sender);
-        address tokenAddress_ = standardizedTokenAddress(tokenId);
-        _deployTokenManager(tokenId, TokenManagerType.MINT_BURN, abi.encode(msg.sender.toBytes(), tokenAddress_));
+        emit CustomTokenIdClaimed(tokenId_, deployer_, salt);
+        if (bytes(destinationChain).length == 0) {
+            _deployTokenManager(tokenId_, tokenManagerType, params);
+        } else {
+            _deployRemoteTokenManager(tokenId_, destinationChain, gasValue, tokenManagerType, params);
```

Comment:
This should check that token manager was already deployed on the current chain

Reply:
_deployTokenManager fails if we try to deploy again. We can check in _deployTokenManager also, but that would introduce a bit more gas for better error messages when trying to deploy duplicates.